api/client/crop_model.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import division
+from builtins import map
+from builtins import zip
+from past.utils import old_div
+import pandas
+import math
+
+from api.client.gro_client import GroClient
+
+
+class CropModel(GroClient):
+    def compute_weights(self, crop_name, metric_name, regions):
+        """Add the weighting data series to this model. Compute the weights,
+        which is the mean value for each region in regions, normalized
+        to add up to 1.0 across regions. Returns a list of weights
+        corresponding to the regions.
+        """
+        # Get the weighting series
+        entities = {
+            'item_id': self.search_for_entity('items', crop_name),
+            'metric_id': self.search_for_entity('metrics', metric_name)
+        }
+        for region in regions:
+            entities['region_id'] = region['id']
+            for data_series in self.get_data_series(**entities):
+                self.add_single_data_series(data_series)
+                break
+        # Compute the average over time for reach region
+        df = self.get_df()
+        
+        def mapper(region):
+            return df[(df['metric_id'] == entities['metric_id']) & \
```

Comment:
should this be indexed by item_id as well, if we do weighting by a different crop?

Reply:
done